instrumentor/runtimemigration/runtimemigration.go

```diff
@@ -0,0 +1,314 @@
+				runtimeDetailsByContainer := workloadInstrumentationConfigReference.Status.RuntimeDetailsByContainer
+
+				for _, containerObject := range sts.Spec.Template.Spec.Containers {
+					err := handleContainerRuntimeDetailsUpdate(
+						containerObject,
+						*originalWorkloadEnvVar,
+						&runtimeDetailsByContainer,
+					)
+					if err != nil {
+						return fmt.Errorf("failed to process container %s in statefulset %s: %v", containerObject.Name, sts.Name, err)
+					}
+				}
+
+				// Update runtimeDetailsByContainer in workloadInstrumentationConfigReference
+				workloadInstrumentationConfigReference.Status.RuntimeDetailsByContainer = runtimeDetailsByContainer
+
+				// Update the InstrumentationConfig status
+				err := clientset.Status().Update(
+					ctx,
+					workloadInstrumentationConfigReference,
+				)
+				if err != nil {
+					return fmt.Errorf("failed to update status for statefulset %s in namespace %s: %v", sts.Name, sts.Namespace, err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func fetchAndProcessDaemonSets(ctx context.Context, clientset client.Client, namespaces map[string]map[string]*v1alpha1.InstrumentationConfig) error {
+	for namespace, workloadNames := range namespaces {
+		var daemonSets appsv1.DaemonSetList
+		err := clientset.List(ctx, &daemonSets, &client.ListOptions{Namespace: namespace})
+		if err != nil {
+			return fmt.Errorf("failed to list daemonsets in namespace %s: %v", namespace, err)
+		}
+
+		for _, ds := range daemonSets.Items {
+			// Checking if the daemonset is in the list of daemonsets that need to be processed
+			if contains(workloadNames, ds.Name) {
+
+				originalWorkloadEnvVar, _ := envoverwrite.NewOrigWorkloadEnvValues(ds.Annotations)
+				workloadInstrumentationConfigReference := workloadNames[ds.Name]
+				runtimeDetailsByContainer := workloadInstrumentationConfigReference.Status.RuntimeDetailsByContainer
+
+				for _, containerObject := range ds.Spec.Template.Spec.Containers {
+					err := handleContainerRuntimeDetailsUpdate(
+						containerObject,
+						*originalWorkloadEnvVar,
+						&runtimeDetailsByContainer)
+					if err != nil {
+						return fmt.Errorf("failed to process container %s in daemonset %s: %v", containerObject.Name, ds.Name, err)
+					}
+				}
+
+				// Update runtimeDetailsByContainer in workloadInstrumentationConfigReference
+				workloadInstrumentationConfigReference.Status.RuntimeDetailsByContainer = runtimeDetailsByContainer
+
+				// Update the InstrumentationConfig status
+				err := clientset.Status().Update(
+					ctx,
+					workloadInstrumentationConfigReference,
+				)
+				if err != nil {
+					return fmt.Errorf("failed to update status for daemonset %s in namespace %s: %v", ds.Name, ds.Namespace, err)
+				}
+			}
+		}
+	}
+	return nil
+}
+func handleContainerRuntimeDetailsUpdate(
+	containerObject v1.Container,
+	originalWorkloadEnvVar envoverwrite.OrigWorkloadEnvValues,
+	runtimeDetailsByContainer *[]v1alpha1.RuntimeDetailsByContainer,
```

Comment:
I don't thinks we should pass a pointer to the slice here since we are not modifying the slice itself but its elements
```suggestion
	runtimeDetailsByContainer []v1alpha1.RuntimeDetailsByContainer,
```